docs/source/extending/contents.rst

```diff
@@ -97,11 +98,11 @@ model. There are three model types: **notebook**, **file**, and **directory**.
    .. _contentfree:
 
    In certain circumstances, we don't need the full content of an entity to
-   complete a Contents API request.  In such cases, we omit the ``mimetype``,
-   ``content``, and ``format`` keys from the model. This most commonly occurs
-   when listing a directory, in which circumstance we represent files within
-   the directory as content-less models to avoid having to recursively traverse
-   and serialize the entire filesystem.
+   complete a Contents API request.  In such cases, we omit the ``content``, and
+   ``format`` keys from the model, and the default values for the ``mimetype``
+   field. This most commonly occurs when listing a directory, in which
```

Comment:
> we omit... and the default values for the `mimetype` field.

This wasn't very clear to me without looking at the extension. Maybe "...and the `mimetype` field may be `None` if the file extension is not recognised."

Reply:
Thanks! I'll add it as a separate sentence to make it clear.

Reply:
Commits added to clarify. The logic is rather convoluted, so the description ends up somewhat unclear, but at least it makes it clear that the value for files might be `None` when doing dir listings.

Reply:
I wonder if we should actually change the logic so it either omits the mimetype entirely when there's no content or uses `application/octet-stream` as the default instead of None. AFAIK, 'octet-stream' is applicable to any regular file including text (but presumably not directories etc.).

Reply:
I'm OK with the way it is currently working, I was just making some assumptions based on the docs that didn't hold up in practice. :)